mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,405 @@
+                )
+                shap_values = explainer(sampled_X)
+            elif raw_model and shap.explainers.Tree.supports_model_with_masker(raw_model, maskers):
+                explainer = shap.explainers.Tree(
+                    raw_model, maskers, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(sampled_X)
+            elif raw_model and shap.explainers.Additive.supports_model_with_masker(
+                    raw_model, maskers
+            ):
+                explainer = shap.explainers.Additive(
+                    raw_model, maskers, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(sampled_X)
+            else:
+                # fallback to default sampling explainer
+                explainer = shap.explainers.Sampling(
+                    predict_fn, X, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(X, sample_rows)
+
+        _logger.info(f'Shap explainer {explainer.__class__.__name__} is used.')
+
+        # TODO: seems infer pip req fail when log_explainer.
+        mlflow.shap.log_explainer(
+            explainer,
+            artifact_path=DefaultEvaluator._gen_log_key('explainer', dataset_name)
+        )
+
+        def plot_summary():
+            pyplot.subplots_adjust(left=0.4)
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_summary, run_id, "shap_summary", dataset_name,
+        )
+
+        def plot_summary_in_js():
+            pyplot.subplots_adjust(left=0.4)
+            shap.summary_plot(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_summary_in_js, run_id, "shap_summary_in_js", dataset_name,
+        )
+
+        def plot_feature_importance():
+            pyplot.subplots_adjust(left=0.4)
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts,
+            temp_dir,
+            plot_feature_importance,
+            run_id,
+            "shap_feature_importance",
+            dataset_name,
+        )
+
+    def _evaluate_classifier(self, temp_dir, model, X, y, dataset_name, feature_names, run_id, evaluator_config):
+        from mlflow.models.evaluation.lift_curve import plot_lift_curve
+
+        # Note: require labels to be number of 0, 1, 2, .. num_classes - 1
+        label_list = sorted(list(set(y)))
+        assert label_list[0] == 0, "Label values must being at '0'."
+        num_classes = len(label_list)
+
+        # TODO: for xgb disable feature names check
+        y_pred = model.predict(X)
+
+        is_binomial = num_classes <= 2
+
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        metrics["accuracy"] = sk_metrics.accuracy_score(y, y_pred)
+        metrics["example_count"] = len(X)
```

Comment:
For `num_classes > 2`, we only seem to output the following information:

```
metrics: {'accuracy': 1.0, 'example_count': 30}
artifacts: {}
```

Is there more content we can add for multi-class classification? Ideally, a plot with per-class ROC curves as an artifact would be helpful. @BenWilson2 can you provide additional suggestions?

Reply:
Can we check if the model can predict probabilities and use the probabilities to compute additional metrics?

Reply:
F1 score
Class vs. Rest individual scores for TP/TN/FP/FN (you just sum up to a binary "is class" and "is not class" to achieve this)
ROC-per-class (using above logic for class vs. rest)
AUC-per-class (using above logic for class vs. rest)
Cohen-Kappa `from sklearn.metrics import cohen_kappa_score`  <- I've only seen statisticians use this
Matthews Correlation Coefficient `from sklearn.metrics import matthews_corrcoef` <- I've only seen statisticians use this
Logloss for the probability scores `from sklearn.metrics import log_loss`

That's about all ya got without rolling some esoteric equation into your eval stack (let's not do that ;) )

Reply:
Thanks @BenWilson2 ! 

F1 score
Class vs. Rest individual scores for TP/TN/FP/FN (you just sum up to a binary "is class" and "is not class" to achieve this)
ROC-per-class (using above logic for class vs. rest)
AUC-per-class (using above logic for class vs. rest)
Logloss for the probability scores from sklearn.metrics import log_loss

^ These all seem like great ideas. @WeichenXu123 can we implement them?

Reply:
A reminder also add confusion matrix for multi-class